lib/fizzy/parser.hpp

```diff
@@ -29,8 +29,8 @@ inline parser_result<uint8_t> parse_byte(const uint8_t* pos, const uint8_t* end)
 template <typename T>
 inline parser_result<T> parse_value(const uint8_t* pos, const uint8_t* end)
 {
-    constexpr auto size = sizeof(T);
-    if (pos + size > end)
+    static constexpr int size = sizeof(T);
```

Comment:
why static?

Reply:
And why `int`, isn't it then signed-unsigned comparison below?

Reply:
Pointer subtraction returns signed integer. I will use `ptrdiff_t` for clarity if easily available.
For `static` I don't remember, I will remove it.